src/github.com/getlantern/flashlight/ui/ui.go

```diff
@@ -62,11 +63,17 @@ func Handle(p string, handler http.Handler) string {
 	return uiaddr + p
 }
 
-func Start(addr string) error {
+func Start(tcpAddr *net.TCPAddr, allowRemote bool) error {
 	var err error
-	l, err = net.Listen("tcp", addr)
-	if err != nil {
-		return fmt.Errorf("Unable to listen at %v: %v", addr, l)
+	// If we want to allow remote connections, we have to bind all interfaces
+	if allowRemote {
+		if l, err = net.Listen("tcp", ":" + strconv.Itoa(tcpAddr.Port)); err != nil {
+			return fmt.Errorf("Unable to listen on all interfaces on port: %d", tcpAddr.Port)
+		}
+	} else {
+		if l, err = net.Listen("tcp", tcpAddr.String()); err != nil {
+			return fmt.Errorf("Unable to listen at %v: %v", tcpAddr.String(), l)
+		}
```

Comment:
Also, it seems like the ListenTCP method is maybe a super minor improvement. I'm not a fan of this passing strings as arguments in general TBH, but that would make it something like:

```go
var addr string
if allowRemote {
    tcpAddr.IP = nil
}
if l, err = net.ListenTCP("ipv4", tcpAddr); err != nil {
    return fmt.Errorf("Unable to listen at %v: %v", tcpAddr, l)
}
```